bands/site_sections/bands.py

```diff
@@ -9,58 +9,70 @@ def index(self, session, id, message=''):
+        band = session.band(id)
+        band_info = session.band_info(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            if not band.performer_count:
+            if not band_info.performer_count:
                 message = 'You must tell us how many people are in your band'
-            elif not band.poc_phone:
+            elif not band_info.poc_phone:
                 message = 'You must enter an on-site point-of-contact cellphone number'
-            elif not band.arrival_time:
+            elif not band_info.arrival_time:
                 message = 'You must enter your expected arrival time'
-            elif band.bringing_vehicle and not band.vehicle_info:
+            elif band_info.bringing_vehicle and not band_info.vehicle_info:
                 message = 'You must provide your vehicle information'
             else:
+                band.info = band_info
+                session.add(band)
+                session.add(band_info)
                 raise HTTPRedirect('index?id={}&message={}', band.id, 'Your band information has been uploaded')
 
         return {
             'band': band,
             'message': message
         }
 
-    def bio(self, session, message='', bio_pic=None, **params):
-        band = session.band(params, restricted=True)
+    def bio(self, session, id, message='', bio_pic=None, **params):
+        band = session.band(id)
+        band_bio = session.band_bio(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            if not band.bio:
+            if not band_bio.bio:
                 message = 'Please provide a brief bio for our website'
 
-            if not message and bio_pic:
-                band.bio_pic_filename = bio_pic.filename
-                band.bio_pic_content_type = bio_pic.content_type.value
-                if band.bio_pic_extension not in c.ALLOWED_BIO_PIC_EXTENSIONS:
+            if not message and bio_pic.filename:
+                band_bio.bio_pic_filename = bio_pic.filename
+                band_bio.bio_pic_content_type = bio_pic.content_type.value
+                if band_bio.bio_pic_extension not in c.ALLOWED_BIO_PIC_EXTENSIONS:
                     message = 'Bio pic must be one of ' + ', '.join(c.ALLOWED_BIO_PIC_EXTENSIONS)
                 else:
-                    with open(band.bio_pic_fpath, 'wb') as f:
+                    with open(band_bio.bio_pic_fpath, 'wb') as f:
                         shutil.copyfileobj(bio_pic.file, f)
 
             if not message:
+                band.bio = band_bio
+                session.add(band)
+                session.add(band_bio)
                 raise HTTPRedirect('index?id={}&message={}', band.id, 'Your bio information has been updated')
 
         return {
             'band': band,
             'message': message
         }
 
-    def w9(self, session, id, message='', w9=None):
+    def w9(self, session, id, message='', w9=None, **params):
```

Comment:
It will no longer be unnecessary once we add the id of the ``BandTaxes`` model as a field.